### DISCUSSION TITLE
Expected Imputer Error

### DISCUSSION THREAD

**Discussion by kylehiroyasu** [alan-turing-institute/sktime#3574] [Q&A]
Hi I am just getting started with `sktime` and I have data where some instances are missing measurements at certain time intervals. I expected that I could just using the `Imputer` to interpolate intermediate values as part of a pipeline. However I get an error when the `Imputer.transform` is called on data size data other than the training data:

```python
import numpy as np
from sktime.transformations.series.impute import Imputer

transformer = Imputer()
train = np.random.uniform(size=(3,10,20))
test = np.random.uniform(size=(5,10,20))
test[2,:,5] = np.nan

transformer.fit(train)
transformer.transform(test)
```

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In [1], line 10
      7 test[2,:,5] = np.nan
      9 transformer.fit(train)
---> 10 transformer.transform(test)

File ~/miniconda3/envs/egg_project/lib/python3.9/site-packages/sktime/transformations/base.py:390, in BaseTransformer.transform(self, X, y)
    387     Xt = self._transform(X=X_inner, y=y_inner)
    388 else:
    389     # otherwise we call the vectorized version of predict
--> 390     Xt = self._vectorize("transform", X=X_inner, y=y_inner)
    392 # convert to output mtype
    393 X_out = self._convert_output(Xt, metadata=metadata)

File ~/miniconda3/envs/egg_project/lib/python3.9/site-packages/sktime/transformations/base.py:930, in BaseTransformer._vectorize(self, methodname, **kwargs)
    927 n_fit = len(self.transformers_.index)
    929 if n != n_fit:
--> 930     raise RuntimeError(
    931         "found different number of instances in transform than in fit. "
    932         f"number of instances seen in fit: {n_fit}; "
    933         f"number of instances seen in transform: {n}"
    934     )
    936 # transform the i-th series/panel with the i-th stored transformer
    937 Xts = []

RuntimeError: found different number of instances in transform than in fit. number of instances seen in fit: 3; number of instances seen in transform: 5
```

Is this the expected behavior? and if so what is the reasoning? Thanks in advance :)

**Comment by fkiraly** [comment]
hm, seems unexpected. Let me convert this to a bug report.

It is expected in cases where the transformer (a) needs to run `fit` and (b) runs by instance, because then in `transform` it won't be fitted to some instances, but as far as I understand this is not the case here. May be a bug.

**Reply by kylehiroyasu** [reply]
Ok, for now I've found a way to get around this issue but I'd be really keen to see this fixed. Thanks for the response and the time you've invested into this project. Seems really interesting and exciting.

**Comment by tmh2606** [comment]
Hello. Was this error addressed? Because I get the same error once I try running MiniRocketMultivariate in the same way.

**Comment by fkiraly** [comment]
Yes, this was addressed - the short summary is, it is not an error but expected behaviour, it was primarily the error message that was unclear about why this was happening. See more extensive discussion here:

https://github.com/sktime/sktime/issues/5606
